source/core_http_client.c

```diff
@@ -553,6 +570,28 @@ static uint32_t getZeroTimestampMs( void )
 
 /*-----------------------------------------------------------*/
 
+static int8_t caseInsensitiveStringCmp( const char * str1,
+                                        const char * str2,
+                                        size_t n )
+{
+    size_t i = 0U;
+
+    assert( str1 != NULL );
```

Comment:
Could either `str1` or `str2` be `NULL` in normal use?  Why are we asserting rather than simply returning a not-equal value if either of these is `NULL`?

The single use of this function already asserts for these cases at lines 2264 & 2269.  Given that a future maintainer might add functionality which uses this function, the documentation for this needs to explicitly say that neither `str1` nor `str2` may be `NULL` and that this is enforced with an `assert()`.

Reply:
In the current use of this function in this file cannot have str1 and str2 as null while in general it can be null, but as this function implementation is specific for our use case and str1 and str2 has already been asserted on the lines as you mentioned, this assert are redundant. I will remove it.